src/main/java/com/limechain/sync/fullsync/FullSyncMachine.java

```diff
@@ -65,28 +65,31 @@ public void start() {
 
         BlockHeader highestFinalizedHeader = blockState.getHighestFinalizedHeader();
         Hash256 stateRoot = highestFinalizedHeader.getStateRoot();
-        accessorHolder.setToStateRoot(stateRoot.getBytes());
 
-        byte[] calculatedMerkleRoot = accessorHolder.getBlockTrieAccessor().getMerkleRoot(StateVersion.V0);
+        BlockTrieAccessor blockTrieAccessor = new BlockTrieAccessor(trieStorage, stateRoot.getBytes());
+
+        byte[] calculatedMerkleRoot = blockTrieAccessor.getMerkleRoot(StateVersion.V0);
         if (!stateRoot.equals(new Hash256(calculatedMerkleRoot))) {
             log.info("State root is not equal to the one in the trie, cannot start full sync");
             return;
         }
 
-        runtime = getRuntimeFromState();
+        runtime = buildRuntimeFromState(blockTrieAccessor);
         blockState.storeRuntime(highestFinalizedHeader.getHash(), runtime);
 
         int startNumber = highestFinalizedHeader
                 .getBlockNumber()
                 .add(BigInteger.ONE)
                 .intValue();
+
         int blocksToFetch = 100;
```

Comment:
This looks likes something that could be a constant.

Reply:
This entire class is far from its final form. I've abstained myself from refactoring beyond my scope, so the changes here are rather necessitated by the desire to polish the `Runtime` class's interface; I'd say dealing with this class any further falls under the umbrella of improving the syncing strategy.

Regarding your concrete comment; it shouldn't even be a constant IMO, but for now it is. 😄